backends/jms/src/main/java/org/hibernate/search/backend/impl/jms/AbstractJMSHibernateSearchController.java

```diff
@@ -69,8 +70,16 @@ public void onMessage(Message message) {
 				log.messageReceivedForUndefinedIndex( indexName );
 				return;
 			}
-			queue = indexManager.getSerializer().toLuceneWorks( (byte[]) objectMessage.getObject() );
-			indexManager.performOperations( queue, null );
+
+			LuceneWorkSerializer luceneWorkSerializer = integrator.getServiceManager().requestService( LuceneWorkSerializer.class );
```

Comment:
The thing is, with the way I envision pieces working together, there is just no need for an index manager to deal with the serializer at all. What I had in mind is this:

* Let index managers deal with actual indexing work: talk to Lucene, talk to ES etc.
* Let backend queue processors deal with propagating work: apply it locally (by going to the IM straight away), or distribute in in the cluster (by sending over JMS/JGroups). It's that latter part where the serializer only is needed. Of course a specific BQP could cache the serializer if it benefits from it.

So bottom line is: One could keep that method, but it's the very nature of the proposed change to remove it as it doesn't really make sense in that proposed design. That said, one could just keep the method on the contract temporarily for the sake of not breaking existing impls.